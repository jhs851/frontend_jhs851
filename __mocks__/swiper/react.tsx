import React from 'react'

const mock: React.FC = ({ children }) => <div>{children}</div>

export default mock
