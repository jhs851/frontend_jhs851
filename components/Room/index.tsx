import React from 'react'
import {
  Body,
  DateText,
  Footer,
  Message,
  Avatar,
  Container,
  StyledRoom,
  UnreadMessageCountContainer,
} from './styled'
import { RelatedRoom as RRoom } from 'types'
import Text from 'components/Text'
import Link from 'next/link'
import useCreatedAt from './hooks/useCreatedAt'

type P = RRoom

const Room: React.FC<P> = ({
  id,
  user,
  local_name,
  unread_message_count,
  last_message,
}) => {
  const hasLastMessage = !!last_message
  const createdTimes = useCreatedAt(last_message?.created_at)

  return (
    <Link href={`/room/${id}`} passHref>
      <StyledRoom data-testid="room-component">
        <Avatar src={user.avatar} alt={local_name} width={56} height={56} />

        <Container>
          <Body>
            <Text size={16} weight={700} letterSpacing={-0.2}>
              {local_name}
            </Text>
            {hasLastMessage && (
              <Message size={13} weight={500} color="coolGrey" ellipsis={true}>
                {last_message.message}
              </Message>
            )}
          </Body>

          {hasLastMessage && (
            <Footer>
              <DateText size={11} letterSpacing={0}>
                {createdTimes}
              </DateText>

              {unread_message_count > 0 && (
                <UnreadMessageCountContainer>
                  <Text
                    size={10}
                    weight={700}
                    color="white"
                    letterSpacing={-0.08}
                  >
                    {unread_message_count}
                  </Text>
                </UnreadMessageCountContainer>
              )}
            </Footer>
          )}
        </Container>
      </StyledRoom>
    </Link>
  )
}

export default Room
