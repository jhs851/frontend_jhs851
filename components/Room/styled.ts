import styled from 'styled-components'
import Image from 'components/Image'
import Text from 'components/Text'
import { flexCenter } from 'styles/flex'

export const StyledRoom = styled.a`
  height: 74px;
  padding: 0 16px;
  background-color: ${({ theme }) => theme.background};
  display: flex;
  align-items: center;
`

export const Avatar = styled(Image)`
  flex: 0 0 56px;
  border-radius: 50%;
  overflow: hidden;
  margin-right: 15px;
`

export const Container = styled.div`
  display: flex;
  flex: 1;
  overflow: hidden;
`

export const Body = styled.div`
  flex: 1;
  overflow: hidden;
`

export const Message = styled(Text)`
  margin-top: 3px;
`

export const Footer = styled.div`
  height: 100%;
  margin-left: 13px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
`

export const DateText = styled(Text)`
  opacity: 0.4;
`

export const UnreadMessageCountContainer = styled.div`
  width: 18px;
  height: 18px;
  margin-top: 6px;
  border-radius: 50%;
  background-color: ${({ theme }) => theme.primary};
  ${flexCenter}
`
