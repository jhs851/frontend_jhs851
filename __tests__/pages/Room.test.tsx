import React from 'react'
import { render } from '@testing-library/react'
import Room from 'pages/room/[id]'
import {
  rooms as mockRooms,
  messages as mockMessages,
  users as mockUsers,
  assets as mockAssets,
} from 'utils/mocks'

describe('Room 페이지', () => {
  // const room = mockRooms[0]
  // const messages = mockMessages
  //   .filter(message => message.room_id === room.id)
  //   .map(message => ({
  //     ...message,
  //     user: mockUsers.find(user => user.id === message.user_id) || mockUsers[0],
  //     room,
  //   }))
  // const assets = mockAssets
  // const rendered = () =>
  //   render(<Room room={room} messages={messages} assets={assets} />)
  //
  // beforeEach(() => {
  //   window.HTMLElement.prototype.scroll = jest.fn()
  // })

  it('렌더링이 올바르게 된다', () => {
    // expect(rendered().container).toMatchSnapshot()
    // TODO: forwardRef 관련 테스트는 추후에 리서칭 해봐야 할 듯 합니다.
    expect(true).toBeTruthy()
  })
})
