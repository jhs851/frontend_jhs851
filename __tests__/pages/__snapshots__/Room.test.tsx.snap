// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room 페이지 렌더링이 올바르게 된다 1`] = `
.c3 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 17px;
  font-weight: 700;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
}

.c11 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 12px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
}

.c16 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  font-weight: 600;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
}

.c7 {
  position: relative;
  overflow: hidden;
}

.c8 {
  height: calc(100vh - 44px);
  padding: 20px 16px 90px;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c8::-webkit-scrollbar {
  display: none;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 10px;
}

.c10 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 1px;
}

.c12 {
  margin: 0 10px;
  opacity: 0.4;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 {
  margin-right: 16px;
}

.c0 {
  height: 44px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: absolute;
  left: 12px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c4 {
  position: absolute;
  right: 12px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c13 {
  margin-top: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c15 {
  padding: 12px;
  border-radius: 12px;
  box-shadow: 0 2px 4px 0;
  word-break: break-word;
}

.c14 {
  margin: 0 4px;
  opacity: 0.4;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px 16px 20px;
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
}

.c19 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 50px;
  margin-right: 12px;
  padding: 17px 16px 16px;
  border-radius: 25px;
  box-shadow: 0 2px 4px 0;
}

.c21 {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  -webkit-transition: opacity 300ms;
  transition: opacity 300ms;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c21:disabled {
  opacity: 0.4;
}

.c20 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  width: 100%;
  resize: none;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c20 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  width: 100%;
  resize: none;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c20 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  width: 100%;
  resize: none;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c20 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  width: 100%;
  resize: none;
  -ms-overflow-style: none;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c20::-webkit-scrollbar {
  display: none;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        data-testid="back-button"
      >
        <div
          class="c2"
        />
      </button>
    </div>
    <p
      class="c3"
      color="white"
      letter-spacing="-0.12"
      size="17"
    >
      장만월 사장님
    </p>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <button
          class="c6"
          data-testid="upload-button"
        >
          <div
            class="c2"
          />
        </button>
        <button>
          <div
            class="c2"
          />
        </button>
      </div>
    </div>
  </header>
  <section
    class="c7"
  >
    <div
      class="c8"
    >
      <div>
        <div
          class="c9"
        >
          <div
            class="c10"
          />
          <p
            class="c11 c12"
            letter-spacing="0"
            size="12"
          >
            2020-05-06
          </p>
          <div
            class="c10"
          />
        </div>
        <div>
          <article
            class="c13"
          >
            <p
              class="c11 c14"
              letter-spacing="0"
              size="12"
            >
              02:34
            </p>
            <div
              class="c15"
            >
              <p
                class="c16"
                color="charcoalGrey2"
                size="14"
              >
                출근했니?
              </p>
            </div>
          </article>
        </div>
        <div>
          <article
            class="c13"
          >
            <p
              class="c11 c14"
              letter-spacing="0"
              size="12"
            >
              02:35
            </p>
            <div
              class="c15"
            >
              <p
                class="c16"
                color="charcoalGrey2"
                size="14"
              >
                출근했냐구?
              </p>
            </div>
          </article>
        </div>
      </div>
      <div>
        <div
          class="c17"
        >
          <div
            class="c10"
          />
          <p
            class="c11 c12"
            letter-spacing="0"
            size="12"
          >
            2020-05-07
          </p>
          <div
            class="c10"
          />
        </div>
        <div>
          <article
            class="c13"
          >
            <p
              class="c11 c14"
              letter-spacing="0"
              size="12"
            >
              02:36
            </p>
            <div
              class="c15"
            >
              <p
                class="c16"
                color="charcoalGrey2"
                size="14"
              >
                어딘데 출근 안하니?
              </p>
            </div>
          </article>
        </div>
      </div>
      <div>
        <div
          class="c17"
        >
          <div
            class="c10"
          />
          <p
            class="c11 c12"
            letter-spacing="0"
            size="12"
          >
            2021-10-24
          </p>
          <div
            class="c10"
          />
        </div>
        <div>
          <article
            class="c13"
          >
            <p
              class="c11 c14"
              letter-spacing="0"
              size="12"
            >
              09:32
            </p>
            <div
              class="c15"
            >
              <p
                class="c16"
                color="charcoalGrey2"
                size="14"
              >
                어딘데 출근 안하니, 죽고싶니?
              </p>
            </div>
          </article>
        </div>
      </div>
      <form
        class="c18"
      >
        <div
          class="c19"
        >
          <textarea
            class="c20"
            name="message"
            placeholder="메시지를 입력하세요.."
            rows="1"
          />
        </div>
        <button
          class="c21"
          disabled=""
          type="submit"
        >
          <div
            class="c2"
          />
        </button>
      </form>
    </div>
  </section>
</div>
`;
